DS-Session-4-Assignment/src/question2/UniqueCharacters.java

```diff
@@ -0,0 +1,69 @@
+package question2;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This class is used to count the number of unique characters in a string and
+ * cache to store previous string result
+ * 
+ * @author admin
+ *
+ */
+
+public class UniqueCharacters {
+
+	private Map<String,Integer> cacheMap;
+	
+	void setCache(Map<String, Integer> cacheMap) {
+		this.cacheMap = cacheMap;
+	}
+	
+	UniqueCharacters() {
+		
+		this.cacheMap=new HashMap<String, Integer>();
+	}
+	
+	/**
+	 * This method to counts the number of unique characters in a string
+	 * @param inputString is the string to be checked
+	 * @return {int} number of unique characters
+	 */
+	int countUniqueCharacters(String inputString){
+		
+		Set<Character> uniqueCharSet=new HashSet<Character>();
+		
+		if(inputString==null || inputString.equals(" ")){
+			
+			throw new AssertionError("input string is invalid");
```

Comment:
Show the message to the user instead of assertion error.

Reply:
Done